### PR TITLE
Add a 1D type independent lookup table.

### PR DESCRIPTION
**Pull request by Cadair**
This is a little ugly, but appears to work, many many tests on their way!!

### CODE REVIEW

File: gwcs/lookup_table.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+    @property
+    def return_units(self):
+        if hasattr(self.lookup_table, 'unit'):
+            return {'y': self.lookup_table.unit}
+        else:
+            return None
+
+    def __init__(self, lookup_table):
+        super().__init__()
```
Comment (nden) [inline gwcs/lookup_table.py]: Add `name=None` to the init and pass it to `super`.

File: gwcs/lookup_table.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+        if not isinstance(lookup_table, u.Quantity):
+            lookup_table = np.asarray(lookup_table)
+
+        self.lookup_table = lookup_table
+
+    def evaluate(self, point):
+        ind = int(np.asarray(np.round(point)))
```
Comment (nden) [inline gwcs/lookup_table.py]: Doesn't this prevent it from evaluating on array of inputs? It looks like this is on purpose but I wonder why.

I think it's good to document the chosen way of rounding the input, as there may be different ways.

File: gwcs/lookup_table.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        return self.lookup_table[ind]
+
+    @property
+    def inverse(self):
+        return _ReverseLookupTable(self.lookup_table)
+
+
+class _ReverseLookupTable(LookupTable):
```
Comment (nden) [inline gwcs/lookup_table.py]: Let' s make this a public class, it's shows up as `model.inverse` anyway.

File: gwcs/lookup_table.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    Parameters
+    ----------
+    lookup_table : `~astropy.units.Quantity` or `numpy.ndarray`
+
+    """
+
+    linear = False
+    fittable = False
+    standard_broadcasting = False
+
```
Comment (nden) [inline gwcs/lookup_table.py]: add the `separable` attribute

File: gwcs/lookup_table.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    inputs = ('x',)
+    outputs = ('y',)
+
+    _inverse = None
+
```
Comment (nden) [inline gwcs/lookup_table.py]: add the `separable` attribute

File: gwcs/lookup_table.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+        else:
+            inds = np.where(self.lookup_table == point)[0]
+
+        if not inds.size:
+            raise ValueError("Value not found in lookup table")
+
+        # There is some strange interaction between inverse and return units,
+        # where this needs to be done here and not with return units.
+        if hasattr(self.lookup_table, 'unit'):
+            inds = u.Quantity(inds, u.pix)
```
Comment (nden) [inline gwcs/lookup_table.py]: Is there a reason to limit this to pixels?

Comment (Cadair) [reply gwcs/lookup_table.py]: not *really* I am just not sure what the best point to ask for the index unit is

**Comment by Cadair** [comment]
@nden I updated it, let me know if I haven't actually answered your queries!

**Comment by Cadair** [comment]
@nden is this good to go? :)

**Comment by Cadair** [comment]
ping? :angel:

**Comment by nden** [comment]
@Cadair Sorry for the delay. I thought it may be a good idea to generalize this a bit to wrap a numpy array but haven't figured out all the details. Then I got pulled in something else. I'll try to come back to this soon.

**Comment by nden** [comment]
The only difference from modeling.Tabular is that this handles string inputs.
Otherwise Tabular can be adapted to not interpolate.

Also, It's unclear how this one handles input arrays, as there's no test, unless I missed it.

**Comment by perrygreenfield** [comment]
@Cadair Can you explain why this needs to be a model? I can see it being a useful tool, but I don't see why it should be a model. A good use case might illustrate how it would be used. But I'm very dubious that it meets the requirements of a model (e.g., that it can be used in a numerical expression). If I see a use case perhaps I can suggest a slightly different approach. It would also be good to know if you heavily depend on any model infrastructure. (E.g., I use the x machinery in modeling to do this work for me).

**Comment by Cadair** [comment]
My original use cases for this have now been encapsulated in `Tabular1D` and coordinate frames.

**Pull request closed without merging**